pkg/operator/controller/controller_service_monitor.go

```diff
@@ -0,0 +1,107 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+
+	operatorclient "github.com/openshift/cluster-dns-operator/pkg/operator/client"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func (r *reconciler) ensureServiceMonitor(dns *operatorv1.DNS, svc *corev1.Service, daemonsetRef metav1.OwnerReference) (*unstructured.Unstructured, error) {
+	desired := desiredServiceMonitor(dns, svc, daemonsetRef)
+
+	current, err := r.currentServiceMonitor(dns)
+	if err != nil {
+		return nil, err
+	}
+
+	if desired != nil && current == nil {
```

Comment:
What about the case where `desired != nil && current != nil`? Should we cover this case and have the client make an `update` call? Something similar to [this](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/controller_router_deployment.go#L39-L42).

Reply:
I see that ^ is not implemented in service monitor for the ingress operator.